sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -326,3 +371,13 @@ class WhileStatementExec(
     body.reset()
   }
 }
+
+class LeaveStatementExec(val label: String) extends LeafStatementExec {
+  var hasBeenMatched: Boolean = false
```

Comment:
can we add some code comments to explain how this flag will be used?

Reply:
I completely forgot to push a commit with comments... Will do during the evening!

Reply:
Done.